### PR TITLE
Serialize connection strings and resote on next app run #26573

### PR DESCRIPTION
**Pull request by salmanalibanani**

### CODE REVIEW

**Comment by salmanalibanani** [comment]
I had to make some changes to how we get a reference to main windown via lifecycle stuff.  Let me know if you discover any side effects, but I think the application behavior is unchanged.

File: PurpleExplorer/Views/ConnectionStringWindow.xaml.cs [hunk 1]
```diff
@@ -28,5 +29,25 @@ public async void btnSendClick(object sender, RoutedEventArgs e)
                 this.Close();
             }
         }
+
+        public void btnSaveConnectionString(object sender, RoutedEventArgs e)
+        {
+            var dataContext = this.DataContext as ConnectionStringWindowViewModel;
+            dataContext.SavedConnectionStrings.Add(dataContext.ConnectionString);
```
Comment (yjlim214) [inline PurpleExplorer/Views/ConnectionStringWindow.xaml.cs:36]: should we only check if the connection string already exists before we add it?

File: PurpleExplorer/ViewModels/MainWindowViewModel.cs [hunk 2]
```diff
@@ -21,8 +25,20 @@ public class MainWindowViewModel : ViewModelBase
         private ServiceBusSubscription _currentSubscription;
         private ServiceBusTopic _currentTopic;
         private Message _currentMessage;
+        private string _connectionString;
+        private IList<string> _savedConnectionStrings;
```
Comment (yjlim214) [inline PurpleExplorer/ViewModels/MainWindowViewModel.cs:29]: maybe we could remove `_savedConnectionStrings` if it's not used?

File: PurpleExplorer/App.xaml.cs [hunk 3]
```diff
@@ -19,18 +21,23 @@ public override void Initialize()
...
-            }
+            var suspension = new AutoSuspendHelper(ApplicationLifetime);
+            RxApp.SuspensionHost.CreateNewAppState = () => new MainWindowViewModel();
+            RxApp.SuspensionHost.SetupDefaultSuspendResume(new NewtonsoftJsonSuspensionDriver("appstate.json"));
+            suspension.OnFrameworkInitializationCompleted();
+            var state = RxApp.SuspensionHost.GetAppState<MainWindowViewModel>();
+
+            new MainWindow { DataContext = state }.Show();
+
+            //if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
```
Comment (yjlim214) [inline PurpleExplorer/App.xaml.cs:34]: maybe we could remove this?

**Review by yjlim214 — COMMENTED** [review]
Looks good! Tested on my local machine and it works well, don't think you introduced any side effects 👍

File: PurpleExplorer/Views/ConnectionStringWindow.xaml.cs (hunk 1 above)
Comment (salmanalibanani) [reply PurpleExplorer/Views/ConnectionStringWindow.xaml.cs]: Done.  Good point, I thought about it and felt may be they want to have some duplicates in there.  Right now, I can't think of why they would want a duplicate!

File: PurpleExplorer/ViewModels/MainWindowViewModel.cs (hunk 2 above)
Comment (salmanalibanani) [reply PurpleExplorer/ViewModels/MainWindowViewModel.cs]: Good catch.  Done

File: PurpleExplorer/App.xaml.cs (hunk 3 above)
Comment (salmanalibanani) [reply PurpleExplorer/App.xaml.cs]: Done.

**Review by yjlim214 — APPROVED** [review]
Looks good!

**Review by armannaj — COMMENTED** [review]
LGTM

**Pull request merged**